src/anchoredinterval.jl

```diff
@@ -160,12 +186,25 @@ function Base.convert(::Type{AnchoredInterval{P}}, interval::Interval{T}) where
 end
 =#
 
-function Base.convert(::Type{AnchoredInterval{Ending}}, interval::Interval{T,L,R}) where {T,L,R}
-    AnchoredInterval{-span(interval), T, L, R}(last(interval))
+_span_fallback(::Type{T}) where T <: TimeType = eps(T)
+_span_fallback(::Type{T}) where T = one(T)
+
+function Base.convert(::Type{AnchoredInterval{Ending}}, interval::Interval{T}) where {T}
```

Comment:
I hadn't seen this before but this is a problematic usage of type parameters and should be removed, if not in this PR than another. First, `Ending` is not a period, second, you can't have an instance of this type, and third, the invariant `convert(T, x)::T` should always hold in every `convert` method.

Reply:
I can agree that this is weird but these end up being very useful (mostly for tests though). The `Ending`/`Beginning` is just used to specify what anchor should be used as otherwise we would otherwise pick one at random. This is worth making a separate issue about but I'm reluctant to make that change here.

Reply:
IMO that should be a constructor argument (or keyword argument), or this should just be a different function.

I understand not wanting to make the change in this PR.

Reply:
https://github.com/invenia/Intervals.jl/issues/109